### ISSUE TITLE
sequence_length in tflearn.lstm(net, 128, sequence_length=?)

### ISSUE THREAD

**Issue by ruizhaogit** [tflearn/tflearn#151]
Hi TFlearn team,

I want to specify the sequence length in lstm for input sequence with dynamic length.

In the lstm layer in tflearn, it says:

"sequence_length: Specifies the length of each sequence in inputs. An int32 or int64 vector (tensor) size `[batch_size]`."

link:https://github.com/tflearn/tflearn/blob/master/tflearn/layers/recurrent.py

And I have a model like this: model.fit(trainX, trainY, n_epoch=2000, shuffle=None, validation_set=(testX, testY), batch_size=1024) 

How can specify the sequence length for both trainX and testX?

For example: my trainX is in form [1000, 300, 150], which has 1000 samples, for each samples there are max 300 timesteps (mostly the latter half is zero), for each time step the input dimension is 150.

The same goes for testX.

Then what will be my sequence_length looks like? A list of 2000 length with int numbers, which specify the number of time steps (sequence length)? Or a two dimentional nparray, which has first column as index and second is the sequnce length?

By the way is there any examples about this sequence length?

When will DNN class support sequence length?

Thank you!

Best,
Rui

**Comment by aymericdamien** [comment]
Hi, unfortunately sequence_length argument is not currently compatible with `DNN` class. So if you want to use it, you need to write your own training process and use a placeholder to feed the sequence length.
We are planning to make dynamic length RNN compatible in the next release (0.3). So it should be available soon :)

**Comment by ruizhaogit** [comment]
Hi @aymericdamien, may I ask how soon will dynamic RNN be compatible? like several weeks? I think only this feature is quite interesting for me. Maybe your team can make a release like (0.2.2) including this feature:) No need to  wait all the other features in (0.3) :)

**Comment by aymericdamien** [comment]
I think within 2 weeks this should be done, maybe faster ;) Yes, actually we will push it to master as soon as it is completed, so no need to wait v0.3.

**Comment by ruizhaogit** [comment]
Sounds cool! Looking forward to using it :+1:

**Comment by aymericdamien** [comment]
Prior to releasing it for TFLearn, I released an example for using dynamic RNN in TensorFlow, might be interesting for you:
https://github.com/aymericdamien/TensorFlow-Examples/blob/master/examples/3_NeuralNetworks/dynamic_rnn.py

**Comment by ruizhaogit** [comment]
Thank you! It looks good! Looking forward to using the TFlearn version LSTM with dynamic sequence length, which has a better compatibility of the current model :)

**Comment by aymericdamien** [comment]
@ruizhaogit Dynamic RNNs are now supported, see:
https://github.com/tflearn/tflearn/blob/master/examples/nlp/dynamic_lstm.py

**Comment by ruizhaogit** [comment]
Cool, Thanks! I will try it ;)
